src/tracks/Pattern.cpp

```diff
@@ -421,6 +421,32 @@ void Pattern::loadSettings( const QDomElement & _this )
 
 
 
+Pattern *  Pattern::previousPattern() const
+{
+	return adjacentPatternByOffset(-1);
+}
+
+
+
+
+Pattern *  Pattern::nextPattern() const
+{
+	return adjacentPatternByOffset(1);
+}
+
+
+
+
+Pattern * Pattern::adjacentPatternByOffset(int offset) const
+{
+	QVector<TrackContentObject *> tcos = m_instrumentTrack->getTCOs();
+	int tcoNum = m_instrumentTrack->getTCONum(this);
+	return dynamic_cast<Pattern*>(tcos.value(tcoNum + offset, NULL));
+}
```

Comment:
@leeavital, I assume you've tested this, but I have to ask... 

How well does this work when the previous or next pattern doesn't exist?   `getTCONum` seems to return `0` and then it could potentially access a non-existing element, right?